src/toast/cc-toast.js

```diff
@@ -0,0 +1,348 @@
+    this.title = null;
+
+    /** @type {Animation} animation controlling the timout and animating the progress bar. */
+    this._animation = null;
+  }
+
+  /* region PRIVATE METHODS*/
+  _dismiss (timeoutExpired) {
+    dispatchCustomEvent(this, 'dismiss', { key: this.key, timeoutExpired });
+  }
+
+  _getIcon () {
+    if (this.intent === 'info') {
+      return infoSvg;
+    }
+    if (this.intent === 'success') {
+      return successSvg;
+    }
+    if (this.intent === 'warning') {
+      return warningSvg;
+    }
+    if (this.intent === 'danger') {
+      return dangerSvg;
+    }
+  }
+
+  _getIconAlt () {
+    if (this.intent === 'info') {
+      return i18n('cc-toast.icon-alt.info');
+    }
+    if (this.intent === 'success') {
+      return i18n('cc-toast.icon-alt.success');
+    }
+    if (this.intent === 'warning') {
+      return i18n('cc-toast.icon-alt.warning');
+    }
+    if (this.intent === 'danger') {
+      return i18n('cc-toast.icon-alt.danger');
+    }
+  }
+
+  _onCloseButtonClick () {
+    this._dismiss(false);
+  }
+
+  _pause () {
+    this._animation?.pause();
+  }
+
+  _resume () {
+    this._animation?.play();
+  }
+  /* endregion*/
+
+  disconnectedCallback () {
+    super.disconnectedCallback();
+    this._animation?.finish();
+    this._animation = null;
+  }
+
+  firstUpdated (_changedProperties) {
+    super.firstUpdated(_changedProperties);
+
+    this._animation = this.shadowRoot.querySelector('.progress-bar-track').animate(
+      [
+        {
+          width: '100%',
+        },
+        {
+          width: 0,
+        },
+      ],
+      {
+        duration: this.timeout,
+        fill: 'forwards',
+        easing: 'linear',
+      },
+    );
+    this._animation.finished.then(() => {
+      this._animation = null;
+      this._dismiss(true);
+    });
+
+    this._animation.play();
+  }
+
+  render () {
+    const tabIndex = (this.timeout > 0) ? '0' : undefined;
+
+    return html`
+      <div class="toast" 
+           @mouseenter=${this._pause} 
+           @mouseleave=${this._resume} 
+           @focus=${this._pause}
+           @blur=${this._resume}
+           tabindex="${ifDefined(tabIndex)}"
+           role="alert">
+        <div class="icon-wrapper">
+          <img class="icon" src="${this._getIcon()}" alt="${this._getIconAlt()}"/>
+        </div>
+        <div class="content">
+          ${this.title ? html`<div class="title-wrapper">${this.title}</div>` : ''}
+          ${this.message ? html`<div class="message-wrapper">${this.message}</div>` : ''}
+        </div>
+        
+        ${this.closeable ? html`
+          <div class="close-button-wrapper">
+            <button
+                class="close-button"
+                @click=${this._onCloseButtonClick}
+                @focus=${this._pause}
+                @blur=${this._resume}
+                title="${i18n('cc-toast.close')}"
+            >
+              <img src="${closeSvg}" alt="${this._getIconAlt()}"/>
```

Comment:
the `alt` of the image is used as the label (accessible name) of the button.
It should describe the action (close) and not the type of toast.

```html
   <img src="${closeSvg}" alt="${i18n('cc-toast.close')}"/>
```